DNN: Required argument "split" not found into dictionary

##### System information (version)
- OpenCV => 4.2.0
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2019

##### Detailed description
I'm trying to inference classification model trained on [Custom Vision](https://www.customvision.ai/) of Azure Cognitive Services using OpenCV DNN module.  
Custom Vision can download trained ONNX model. The following error occurs when read net from it.  

```
OpenCV: terminate handler is called! The last OpenCV error is:
OpenCV(4.2.0) Error: Requested object was not found (Required argument "split" not found into dictionary) in cv::dnn::dnn4_v20191202::Dict::get, file C:\opencv-4.2.0\modules\dnn\include\opencv2/dnn/dnn.inl.hpp, line 349
```

This is part of model visualize image using Netron.
![2020-01-17_10h51_43](https://user-images.githubusercontent.com/816705/72579738-59b39e80-391d-11ea-851e-2cc422be10b8.png)
(Other model visualize image is [here](https://user-images.githubusercontent.com/816705/72579941-e8282000-391d-11ea-8bee-fbfc2989c6e3.png). All model visualize image is [here](https://user-images.githubusercontent.com/816705/72579836-98e1ef80-391d-11ea-9b8e-60a2aed95f35.png).)

What do you think? Thanks,

##### Steps to reproduce
This issue can be reproduced with the following code: 

```cpp
const std::string model = "./model.onnx";
cv::dnn::Net net = cv::dnn::readNet( model ); // error
```

Hi @UnaNancyOwen @alalek 
I think I am able to reproduce the error. 
**Code**
```cpp
// opencv_debug.cpp
#include<opencv2/dnn.hpp>
int main()
{
    const std::string model = "./model.onnx";
    cv::dnn::Net net = cv::dnn::readNet( model ); // error
}
 ```
I used, `g++ opencv_debug.cpp -o issue_16370 -lopencv_dnn -lopencv_core` and got the following,
```cpp
terminate called after throwing an instance of 'cv::Exception'  what():  OpenCV(3.4.9-dev) /home/czgdp1807/opencv_project/opencv/modules/dnn/include/opencv2/dnn/dnn.inl.hpp:333: error: (-204:Requested object was not found) Required argument "split" not found into dictionary in function 'get'Aborted (core dumped)
```
Can I start working on this issue as it appears that no PR has referenced this issue.

--- a/modules/dnn/src/onnx/onnx_importer.cpp
+++ b/modules/dnn/src/onnx/onnx_importer.cpp
@@ -398,6 +398,7 @@ void ONNXImporter::populateNet(Net dstNet)
         std::string layer_type = node_proto.op_type();
         layerParams.type = layer_type;
 
+
         if (layer_type == "MaxPool")
         {
             layerParams.type = "Pooling";
@@ -482,7 +483,7 @@ void ONNXImporter::populateNet(Net dstNet)
          }
         else if (layer_type == "Split")
         {
-            if(layerParams.has("split"))
+            if (layerParams.has("split"))
             {
                 DictValue splits = layerParams.get("split");
                 const int numSplits = splits.size();
@@ -494,13 +495,12 @@ void ONNXImporter::populateNet(Net dstNet)
                     slicePoints[i] = slicePoints[i - 1] + splits.get<int>(i - 1);
                 }
                 layerParams.set("slice_point", DictValue::arrayInt(&slicePoints[0], slicePoints.size()));
-                layerParams.type = "Slice";
             }
             else
             {
                 layerParams.set("num_split", node_proto.output_size());
-                layerParams.type = "Slice";
             }
+            layerParams.type = "Slice";
         }
         else if (layer_type == "Add" || layer_type == "Sum")
         {

--- a/modules/dnn/src/onnx/onnx_importer.cpp
+++ b/modules/dnn/src/onnx/onnx_importer.cpp
@@ -398,7 +398,6 @@ void ONNXImporter::populateNet(Net dstNet)
         std::string layer_type = node_proto.op_type();
         layerParams.type = layer_type;
 
-
         if (layer_type == "MaxPool")
         {
             layerParams.type = "Pooling";
@@ -483,17 +482,25 @@ void ONNXImporter::populateNet(Net dstNet)
          }
         else if (layer_type == "Split")
         {
-            DictValue splits = layerParams.get("split");
-            const int numSplits = splits.size();
-            CV_Assert(numSplits > 1);
-
-            std::vector<int> slicePoints(numSplits - 1, splits.get<int>(0));
-            for (int i = 1; i < splits.size() - 1; ++i)
+            if(layerParams.has("split"))
             {
-                slicePoints[i] = slicePoints[i - 1] + splits.get<int>(i - 1);
+                DictValue splits = layerParams.get("split");
+                const int numSplits = splits.size();
+                CV_Assert(numSplits > 1);
+
+                std::vector<int> slicePoints(numSplits - 1, splits.get<int>(0));
+                for (int i = 1; i < splits.size() - 1; ++i)
+                {
+                    slicePoints[i] = slicePoints[i - 1] + splits.get<int>(i - 1);
+                }
+                layerParams.set("slice_point", DictValue::arrayInt(&slicePoints[0], slicePoints.size()));
+                layerParams.type = "Slice";
             }
-            layerParams.set("slice_point", DictValue::arrayInt(&slicePoints[0], slicePoints.size()));
-            layerParams.type = "Slice";
+            else
+            {
+                layerParams.set("num_split", node_proto.output_size());
+                layerParams.type = "Slice";
+            }
         }
         else if (layer_type == "Add" || layer_type == "Sum")
         {

--- a/modules/dnn/test/test_onnx_importer.cpp
+++ b/modules/dnn/test/test_onnx_importer.cpp
@@ -14,7 +14,7 @@ namespace opencv_test { namespace {
 template<typename TString>
 static std::string _tf(TString filename, bool required = true)
 {
-    return findDataFile(std::string("dnn/onnx/") + filename, required);
+    return findDataFile(std::string("../../opencv_extra/testdata/dnn/onnx/") + filename, required);
 }
 
 class Test_ONNX_layers : public DNNTestLayer
@@ -388,6 +388,10 @@ TEST_P(Test_ONNX_layers, ReduceL2)
 
 TEST_P(Test_ONNX_layers, Split)
 {
+    if (backend == DNN_BACKEND_INFERENCE_ENGINE_NN_BUILDER_2019)
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_IE_NN_BUILDER);
+    if (backend == DNN_BACKEND_INFERENCE_ENGINE_NGRAPH)
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_IE_NGRAPH);
     testONNXModels("split_1");
     testONNXModels("split_2");
     testONNXModels("split_3");

--- a/modules/dnn/test/test_onnx_importer.cpp
+++ b/modules/dnn/test/test_onnx_importer.cpp
@@ -388,7 +388,10 @@ TEST_P(Test_ONNX_layers, ReduceL2)
 
 TEST_P(Test_ONNX_layers, Split)
 {
-    testONNXModels("split");
+    testONNXModels("split_1");
+    testONNXModels("split_2");
+    testONNXModels("split_3");
+    testONNXModels("split_4");
 }
 
 TEST_P(Test_ONNX_layers, Slice)

--- a/modules/dnn/test/test_onnx_importer.cpp
+++ b/modules/dnn/test/test_onnx_importer.cpp
@@ -14,7 +14,7 @@ namespace opencv_test { namespace {
 template<typename TString>
 static std::string _tf(TString filename, bool required = true)
 {
-    return findDataFile(std::string("../../opencv_extra/testdata/dnn/onnx/") + filename, required);
+    return findDataFile(std::string("dnn/onnx/") + filename, required);
 }
 
 class Test_ONNX_layers : public DNNTestLayer

--- a/modules/dnn/test/test_onnx_importer.cpp
+++ b/modules/dnn/test/test_onnx_importer.cpp
@@ -386,6 +386,11 @@ TEST_P(Test_ONNX_layers, ReduceL2)
     testONNXModels("reduceL2");
 }
 
+TEST_P(Test_ONNX_layers, Split)
+{
+    testONNXModels("split");
+}
+
 TEST_P(Test_ONNX_layers, Slice)
 {
 #if defined(INF_ENGINE_RELEASE) && INF_ENGINE_VER_MAJOR_LT(2019010000)

